server/web/routes/proposals.js

```diff
@@ -322,6 +323,51 @@ async function getObjectFromS3(fileName) {
   });  
 }
 
+async function findAllProposalsInSameGroup(proposal, includeSelf = false) {
+  const history = [];
+  let original = null;
+  if (proposal.hasChild || proposal.parentId) {
+    // find all revised proposals
+    const revisedProposal = await Proposal.find({
+      groupId: proposal.groupId ? proposal.groupId : proposal._id.toString(),
+    });
```

Comment:
You can still keep this. if proposal doesn't have any revised proposals uploaded, then, this list, revisedProposals is going to be empty. Also since it is a list, please rename the variable to revisedProposals (prular).